arroyo/backends/kafka/consumer.py

```diff
@@ -182,6 +183,11 @@ def __init__(
                 "invalid value for 'enable.auto.offset.store' configuration"
             )
 
+        self.__incremental_cooperative = incremental_cooperative
+
+        if self.__incremental_cooperative is True:
+            configuration["partition.assignment.strategy"] = "cooperative-sticky"
```

Comment:
Why providing `incremental_cooperative` as a dedicated flag instead of passing it through the `configuration` mapping like all the others config parameters ?

Reply:
I had it the other way originally but figured it was easier to pass a boolean than remember this string then check for it to decide whether to apply incremental_assign. Thinking about it again, I might switch back to avoid changing the interface of this class though.

Reply:
> I might switch back to avoid changing the interface of this class though.

Also we would not reintroduce two separate ways to provide Kafka config that took us long to clean up

Reply:
I am talking about all those fields we were passing via CLI that we are slowly moving to settings.

Reply:
Unfortunately passing via CLI is still the easiest option currently as we don't have a mechanism yet to provide different settings for consumers/producers of a topic via settings.